objects/src/accounts/mod.rs

```diff
@@ -432,12 +324,10 @@ mod tests {
             .collect();
         let storage = AccountStorage::new(slot_items, vec![]).unwrap();
 
-        // create account
-        let id = AccountId::try_from(ACCOUNT_ID_REGULAR_ACCOUNT_IMMUTABLE_CODE_ON_CHAIN).unwrap();
         Account::new(id, vault, storage, code, nonce)
     }
 
-    fn build_account_delta(
+    pub fn build_account_delta(
         added_assets: Vec<Asset>,
         removed_assets: Vec<Asset>,
         nonce: Felt,
```

Comment:
I think looking at the issue and context regarding this code originally (#423), this could receive the storage changes as a parameter and then changes could be validated outside as needed.